sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,436 @@
+
+client.begin_labeled_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> LabeledCustomModel
+
+client.begin_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> CustomModel
+
+client.begin_extract_form(form: Any, model_id: str,) -> LROPoller -> List[ExtractedPage]
+
+client.begin_extract_labeled_fields(form: Any, model_id: str) -> LROPoller -> List[LabeledExtractedPage]
+
+client.list_custom_models() -> ItemPaged[ModelInfo]
+
+client.get_models_summary() -> ModelsSummary
+
+client.delete_custom_model(model_id: str) -> None
+```
+
+### Custom Models Unlabeled
+```python
+# Training ---------------------------------------------------
+class CustomModel:
+    model_id: str
+    status: str
+    created_date_time: ~datetime.datetime
+    last_updated_date_time: ~datetime.datetime
+    form_clusters: dict[str, list[str]]
+    train_result: TrainResult
+
+class TrainResult:
+    documents: List[TrainingDocumentInfo]
+    average_model_accuracy: float
+    training_errors: List[ErrorInformation]
+
+class TrainingDocumentInfo:
+    document_name: str
+    status: str
+    pages: int
+    document_errors: List[ErrorInformation]
+
+class ErrorInformation:
+    code: str
+    message: str
+
+# Analyze ---------------------------------------------------
+class ExtractedPage:
+    fields: List[ExtractedField]
+    tables: List[ExtractedTable]
+    page_number: int
+    cluster_id: int
+
+class ExtractedField:
+    name: TextValue
+    value: TextValue
+    confidence: float
+
+class TextValue:
+    text: str
+    bounding_box: List[float]
+    raw_field: List[ExtractedLine]
+
+class ExtractedLine:
+    text: str
+    bounding_box: List[float]
+    language: str
+    words: List[ExtractedWord]
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+```
+
+### Custom Models Labeled
+```python
+# Training ---------------------------------------------------
+class LabeledCustomModel:
+    model_id: str
+    status: str
+    created_date_time: ~datetime.datetime
+    last_updated_date_time: ~datetime.datetime
+    train_result: LabeledTrainResult
+
+class LabeledTrainResult:
+    documents: List[TrainingDocumentInfo]
+    fields: List[FieldNames]
+    average_model_accuracy: float
+    training_errors: List[ErrorInformation]
+
+class TrainingDocumentInfo:
+    document_name: str
+    status: str
+    pages: int
+    document_errors: List[ErrorInformation]
+
+class FieldNames:
+    field_name: str
+    accuracy: float
+
+class ErrorInformation:
+    code: str
+    message: str
+
+# Analyze ---------------------------------------------------
+class LabeledExtractedPage:
+    fields: List[ExtractedLabel]
+    tables: List[ExtractedTable]
+    page_number: int
+
+class ExtractedLabel:
+    name: str
+    value: LabelValue
+
+class LabelValue:
+    text: str
```

Comment:
A `LabelValue` is technically a `FieldValue` which can have typed fields. Today they are all strings (for custom labeled), but we should leave room for types.

Reply:
That's good to know. Will sync with team about this so we leave room for types